pyvista/core/utilities/features.py

```diff
@@ -86,16 +108,52 @@ def voxelize(mesh, density=None, check_surface=True):
     # indexing='ij' is used here in order to make grid and ugrid with x-y-z ordering, not y-x-z ordering
     # see https://github.com/pyvista/pyvista/pull/4365
 
-    # Create unstructured grid from the structured grid
-    grid = pyvista.StructuredGrid(x, y, z)
-    ugrid = pyvista.UnstructuredGrid(grid)
+    if vol:
```

Comment:
If we keep this as one function, we should move the above code that is specific to a mode into the else block below.